tests/foreman/ui/test_bookmarks.py

```diff
@@ -0,0 +1,516 @@
+        'setup': entities.HostGroup},
+    {'name': 'DiscoveryRule', 'controller': 'discovery_rules'},
+    {'name': 'GlobalParameter', 'controller': 'common_parameters',
+        'setup': entities.CommonParameter},
+    {'name': 'ConfigGroup', 'controller': 'config_groups',
+        'setup': entities.ConfigGroup},
+    {'name': 'PuppetEnv', 'controller': 'environments',
+        'setup': entities.Environment},
+    {'name': 'PuppetClass', 'controller': 'puppet_classes',
+        'setup': entities.PuppetClass},
+    {'name': 'SmartVariable', 'controller': 'smart_variables',
+        'setup': entities.SmartVariable},
+    {'name': 'SmartProxy', 'controller': 'smart_proxies'},
+    {'name': 'ComputeResource', 'controller': 'compute_resources',
+        'setup': entities.DockerComputeResource},
+    {'name': 'ComputeProfile', 'controller': 'compute_profiles'},
+    {'name': 'Subnet', 'controller': 'subnets',
+        'setup': entities.Subnet},
+    {'name': 'Domain', 'controller': 'domains',
+        'setup': entities.Domain},
+    {'name': 'Realm', 'controller': 'realms',
+        'setup': entities.Realm},
+    {'name': 'Location', 'controller': 'locations'},
+    {'name': 'Organization', 'controller': 'organizations'},
+    {'name': 'User', 'controller': 'users'},
+    {'name': 'UserGroup', 'controller': 'user_groups',
+        'setup': entities.UserGroup},
+    {'name': 'Role', 'controller': 'roles'},
+    {'name': 'SettingTestCase', 'controller': 'settings'},
+]
+
+
+def create_entity(entity):
+    """Take entity name and call nailgun methods to create a random instance
+
+    :param entity: Nailgun entity class to be processed
+    """
+
+    ng_entity = entity()
+    ng_entity.create_missing()
+    ng_entity.create()
+
+
+class BookmarkTestCase(UITestCase):
+    """Test for common Bookmark operations in UI"""
+
+    # CREATE TESTS
+    @stubbed()
+    @tier1
+    def test_positive_create_bookmark_populate_auto(self):
+        """Test creating a bookmark with auto-populating of the query
+        @Feature: Scoped Search Bookmark Create
+
+        @Steps:
+
+        1. Navigate to the entity page
+        2. Input a random text into the search field
+        3. Choose "bookmark this search" from the search drop-down menu
+        4. Input a random name for a bookmark name
+        5. Verify the query field is automatically populated and the public
+        option is checked
+        6. Click the create button
+        7. Verify that bookmark's name appears in the search dropdown
+        8. List the bookmarks (Navigate to Administer -> Bookmarks)
+
+        @Assert: No errors, Bookmark is displayed, controller matches the
+        entity the bookmark was created for
+
+        @Status: Manual
+        """
+
+        for entity in entity_list:
+            if entity.get('setup'):
+                create_entity(entity['setup'])
+            pass
+
+    @stubbed()
+    @tier1
+    def test_positive_create_bookmark_2(self):
```

Comment:
We stopped numbering the test cases, instead please add some meaning description